src/main/resources/com/typesafe/sbt/packager/archetypes/scripts/ash-template

```diff
@@ -129,4 +129,4 @@ java_cmd="$(get_java_cmd)"
 # If a configuration file exist, read the contents to $opts
 [ -f "$script_conf_file" ] && opts=$(loadConfigFile "$script_conf_file")
 
-eval "exec $java_cmd $java_opts -classpath $app_classpath $opts $app_mainclass $app_commands $residual_args"
+exec $java_cmd $java_opts -classpath $app_classpath $opts $app_mainclass $app_commands $residual_args
```

Comment:
I don't know why `eval` was used here. It breaks `app_mainclass` if it contains `$` symbols, so I've removed it and it doesn't seem to break any tests, so… I guess it's fine?

Reply:
#1334 added `checkComplexResidual` but never called it from the [test file](https://github.com/sbt/sbt-native-packager/blob/9e009b0acd9f5d83c58174052ce9511293399047/src/sbt-test/ash/command-line-settings/test). If we fix that, hopefully we'll see why `eval` was added.

Reply:
OK, I've added that to the test file. Ironically, the tests still pass, and they fail if I put the `eval`  back in. I think everything here is broken, both the way "residual args" are being handled and the test that is supposed to test them.

Reply:
I agree, I'm not confident that those tests work. Digging into the blame, there's a lot of history for that line of code going back to #1334, #1303, #1095 and before.

If you call `myProgram --param1 "value1 value2"`, does it correctly pass two arguments (arg[0]="--param1", arg[1]="value1 value2")?

Reply:
I've now fixed the tests and the escaping of residual args, so it should all be passed through correctly now. In fact I've added tests for all kinds of nasty things that might confuse a shell script and they're all passed through correctly.